haas/cli.py

```diff
@@ -79,17 +79,38 @@ def object_url(*args):
         url += '/' + urllib.quote(arg,'')
     return url
 
+def do_request(fn, url, data={}):
+    """Helper function for making HTTP requests against the API.
+
+    Arguments:
+
+        `fn` - a function from the requests library, one of requests.put,
+               requests.get...
+        `url` - The url to make the request to
+        `data` - the body of the request.
+
+    If the environment variables HAAS_USERNAME and HAAS_PASSWORD are
+    defined, The request will use HTTP basic auth to authenticate, with
+    the given username and password.
+    """
+    kwargs = {}
+    username = os.getenv('HAAS_USERNAME')
+    password = os.getenv('HAAS_PASSWORD')
+    if username is not None and password is not None:
```

Comment:
It is possible to set it to an empty string (e.g. `HAAS_PASSWORD= haas create_headnode ...`), but the environment variable must be defined. If you look at `docs/rest_api.md`, you'll see that the password field in the body of the request for `user_create` is not optional, so there's not actually a way to create a passwordless user, even on the server side. This is unchanged from before (remember, these calls *did* exist, we just weren't really using the user objects for anything, so it didn't matter).